src/browserlib/extract-dfns.mjs

```diff
@@ -122,6 +122,47 @@ function isNotAlreadyExported(dfn, idx, list) {
   return first === dfn;
 }
 
+// Extract the element's inner HTML content, removing any complex structure,
+// so that the result can be injected elsewhere without creating problems.
+function getHtmlProseDefinition(proseEl) {
+  // Apply modifications to a copy of the element
+  proseEl = proseEl.cloneNode(true);
+
+  // Drop asides that authoring tools add here and there
+  let el;
+  const asideSelector = [
+    'aside', '.mdn-anno', '.wpt-tests-block', '.annotation',
+    '[id^=dfn-panel-]'
+  ].join(',');
+  while (el = proseEl.querySelector(asideSelector)) {
+    el.remove();
+  }
+
+  // Keep simple grouping content and text-level semantics elements
+  const keepSelector = [
+    'blockquote', 'dd', 'div', 'dl', 'dt', 'figcaption', 'figure', 'hr', 'li',
```

Comment:
while having a `<dl>` inside the definition block might work, I don't think it would work as a container of a definition - maybe we should have distinct selectors for validating the content vs validating the container? the same would probably apply with `<ol>`, `<ul>`, and `<table>` if it was added.

Reply:
Could you clarify what "container" means here?

I would define it as the element with the `data-defines` attribute. And the code does not check the container at all in practice. It can be anything, the code will strip it away and only report its contents in the `prose` (or `htmlProse`) property.

Reply:
you're right, I managed to confuse myself on this